chain/epoch-manager/src/adapter.rs

```diff
@@ -143,6 +144,20 @@ pub trait EpochManagerAdapter: Send + Sync {
         block_height: BlockHeight,
         approvals: &[Option<Signature>],
     ) -> Result<bool, Error>;
+
+    /// Verify approvals and check threshold, but ignore next epoch approvals and slashing
+    fn verify_approvals_and_threshold_orphan(
+        &self,
+        epoch_id: &EpochId,
+        can_approved_block_be_produced: &dyn Fn(
+            &[Option<Signature>],
+            &[(Balance, Balance, bool)],
```

Comment:
let's explain the arguments:

first balance - is stake this epoch
second balance - stake next epoch
bool - is slashed